web/profiles/custom/yalesites_profile/modules/custom/ys_themes/ys_themes.module

```diff
@@ -158,3 +158,39 @@ function ys_themes_build_css_variables() : string {
   $css[] = '}';
   return implode(PHP_EOL, $css);
 }
+
+/**
+ * Implements hook_form_alter().
+ *
+ * Adds a placeholder to the search form and visually hides the submit button.
+ */
+function ys_themes_form_alter(&$form, &$form_state, $form_id) {
+  if ($form_id == "views_exposed_form" && $form['#id'] == "views-exposed-form-search-page-1") {
+    $form['keywords']['#attributes'] = ['placeholder' => [t("Search this site")]];
+    $form['actions']['submit']['#attributes']['class'][] = "visually-hidden";
+  }
+}
+
+/**
+ * Implements hook_theme_suggestions_form_element_label_alter().
+ *
+ * Add a template to visually hide the search form label and add custom icon.
+ */
+function ys_themes_theme_suggestions_form_element_label_alter(array &$suggestions, array $variables, $hook) {
+  if (isset($variables['element']['#id']) && strpos($variables['element']['#id'], "edit-keywords") === 0) {
+    $suggestions[] = $hook . '__search_form';
+  }
+}
```

Comment:
I tend to dislike hooks, but they are effective especially when trying to tweak small things without having to recreate something that is in core. None of these things are problematic. But if this list grows, that I would rather create our own block without own template. 

Why all the hate? Who hurt me? It's things like looking at the string position for a specific string. Or checking an autogenerated form id. If these strings change (possibly as part of a core update or installing a third party module) then our code stops firing. 

So my gut says to leave this. It is effective and an appropriate use of Drupal's hooks. But I will also pitch the question over to @ModulesUnraveled. Do you often create a custom search block in Drupal so that you can control the form? Are we near that point or is this just standard theming business?

Reply:
This is how I've always done it. No one has ever suggested creating a custom block... so I don't have any preference at this time.

Reply:
Thank you for explaining the reasons why hooks can be problematic - I was curious and I understand now. I tried to think of another way we can alter the search form and Drupal doesn't provide template suggestions for this form - so we'd still have to write a template suggestion based on the exposed form name (views-exposed-form-search-page-1) and then also override other templates as well to get the placeholder and ```visually-hidden``` class in there too... so it is possible to do. For now, you said let's ship this version and if we need to later we can refactor?